pymongo/mongo_client.py

```diff
@@ -1600,9 +1602,11 @@ def _process_kill_cursors(self):
     def _process_periodic_tasks(self):
         """Process any pending kill cursors requests and
         maintain connection pool parameters."""
-        self._process_kill_cursors()
         try:
+            self._process_kill_cursors()
             self._topology.update_pool(self.__all_credentials)
+        except InvalidOperation:
+            pass
```

Comment:
We only want to ignore this error if the client was closed.

Reply:
Done.